libs/vkd3d/resource.c

```diff
@@ -3494,6 +3494,18 @@ static void d3d12_descriptor_heap_write_null_descriptor_template(struct d3d12_de
     VkDeviceAddress *va;
 
     heap = desc->heap;
+
+    /* When mutable descriptors are not supported, set a dummy type.
+       This will make those drivers not care about the null type being different between
+       null writes. */
+    if (!heap->has_mutable_descriptors)
```

Comment:
I was thinking of placing this in heap->null_descriptor_template to keep the pointer chasing down to a bare minimum since we already load the set_mask in same cache line. The alternative is just to chase desc->heap->device->blah every time, but that seems a bit inoptimal since this is a performance PR after all.